spec/features_spec.rb

```diff
@@ -0,0 +1,207 @@
+    score.record_roll(5)
+    score.record_roll(2)
+    score.record_roll(7)
+    score.record_roll(3)
+    score.record_roll(3)
+    score.record_roll(7)
+    score.record_roll(10)
+    score.record_roll(2)
+    score.record_roll(1)
+    score.record_roll(6)
+    score.record_roll(1)
+    score.record_roll(2)
+    score.record_roll(4)
+    score.record_roll(3)
+    score.record_roll(7)
+    score.record_roll(10)
+    score.record_roll(2)
+    score.record_roll(1)
+    expect{score.record_roll(2)}.to raise_error("That was the last frame, kid. You want to roll again, start a new game")
+  end
+
+  it 'does not allow more than one bonus roll if a spare is rolled' do
+    score.record_roll(5)
+    score.record_roll(2)
+    score.record_roll(7)
+    score.record_roll(3)
+    score.record_roll(3)
+    score.record_roll(7)
+    score.record_roll(10)
+    score.record_roll(2)
+    score.record_roll(1)
+    score.record_roll(6)
+    score.record_roll(1)
+    score.record_roll(2)
+    score.record_roll(4)
+    score.record_roll(3)
+    score.record_roll(7)
+    score.record_roll(10)
+    score.record_roll(2)
+    score.record_roll(8)
+    score.record_roll(10)
+    expect(score.calculate_score).to eq(129)
+    expect{score.record_roll(2)}.to raise_error("That was the last frame, kid. You want to roll again, start a new game")
+  end
+
+  it 'does not allow more than two bonus rolls if a strike is rolled' do
+    score.record_roll(5)
+    score.record_roll(2)
+    score.record_roll(7)
+    score.record_roll(3)
+    score.record_roll(3)
+    score.record_roll(7)
+    score.record_roll(10)
+    score.record_roll(2)
+    score.record_roll(1)
+    score.record_roll(6)
+    score.record_roll(1)
+    score.record_roll(2)
+    score.record_roll(4)
+    score.record_roll(3)
+    score.record_roll(7)
+    score.record_roll(10)
+    score.record_roll(2)
+    score.record_roll(1)
+    expect{score.record_roll(2)}.to raise_error("That was the last frame, kid. You want to roll again, start a new game")
+  end
+  it 'does not allow more than one bonus roll if a spare is rolled' do
+    score.record_roll(5)
+    score.record_roll(2)
+    score.record_roll(7)
+    score.record_roll(3)
+    score.record_roll(3)
+    score.record_roll(7)
+    score.record_roll(10)
+    score.record_roll(2)
+    score.record_roll(1)
+    score.record_roll(6)
+    score.record_roll(1)
+    score.record_roll(2)
+    score.record_roll(4)
+    score.record_roll(3)
+    score.record_roll(7)
+    score.record_roll(10)
+    score.record_roll(2)
+    score.record_roll(8)
+    score.record_roll(10)
+    expect(score.calculate_score).to eq(129)
+    expect{score.record_roll(2)}.to raise_error("That was the last frame, kid. You want to roll again, start a new game")
+  end
+
+  it 'does not allow more than two bonus rolls if a strike is rolled' do
+    expect{ 13.times{score.record_roll(10)} }.to raise_error("That was the last frame, kid. You want to roll again, start a new game")
+  end
+end
```

Comment:
Really thorough testing, I feel like you covered all the weird little aspects of bowling that might cause hiccups and stuff, plus edge cases